### PR TITLE
Fixes fedimint/minimint#81 off by one in blind signature aggregation …

### PR DESCRIPTION
**Pull request by jkitman**
Fix for the off by one in blind signature aggregation code.

I enabled the commented-out tests in `modules/minimint-mint/src/lib.rs` that also tests this issue.

### CODE REVIEW

**Comment by elsirion** [comment]
Just did the rebase on master and squashed the fixups. Should be merge ready afterwards.

File: modules/minimint-mint/src/lib.rs [hunk 1]
```diff
@@ -401,14 +389,28 @@ impl Mint {
...
+                Ok((msg.0, blind_signature))
+            },
+        )?))
+    }
+
     fn combine(
         &self,
         partial_sigs: Vec<(PeerId, PartialSigResponse)>,
     ) -> (Result<SigResponse, CombineError>, MintShareErrors) {
         // FIXME: decide on right boundary place for this invariant
```
Comment (elsirion) [inline modules/minimint-mint/src/lib.rs:407]: This FIXME belongs below your insert, it relates to the duplicate sig shares per peer.

File: modules/minimint-mint/src/lib.rs [hunk 2]
```diff
@@ -263,43 +257,37 @@ impl FederationModule for Mint {
...
+                                peer_id: peer,
+                            })
+                        }));
+
+                        batch_tx.append_insert(OutputOutcomeKey(issuance_id), blind_signature);
+                        batch_tx.commit();
+                        Some(batch)
+                    }
+                    Err(e) => {
+                        if e != CombineError::TooFewShares {
```
Comment (elsirion) [inline modules/minimint-mint/src/lib.rs:286]: Maybe make the other case a trace after removing the debug output `"Trying to combine sig shares for issuance request {}",`

**Review by elsirion — APPROVED** [review]
Looks great! Thx for also refactoring the code a bit. Just applying a few fixes to my nits and the I'll merge.

File: modules/minimint-mint/src/lib.rs (hunk 1 above)
Comment (jkitman) [reply modules/minimint-mint/src/lib.rs]: Ah okay, I was unsure what that comment meant.

**Comment by elsirion** [comment]
Thx a lot!

**Pull request merged**